src/com/oltpbenchmark/api/Worker.java

```diff
@@ -592,6 +565,10 @@ public final void run() {
                             assert (false) : String.format("Unexpected status '%s' for %s", status, next);
                     } // SWITCH
                 }
+                // Don't retry if it is not a NewOrder Transaction.
+                if (!next.getName().equals("NewOrder")) {
```

Comment:
lets use this style of check, as its safer: https://github.com/yugabyte/tpcc/blob/master/src/com/oltpbenchmark/api/Worker.java#L496

Reply:
Done.